### DISCUSSION TITLE
Improve the current ADR system

### DISCUSSION THREAD

**Discussion by rmontagnin** [desmos-labs/desmos#690] [General]
Today I was taking a look at the ADRs that we have merged and the ones that are currently opened as PRs and I think we have the following problems: 

- All the modules we have inside the `staging` folder do not have an ADR (`x/posts`, `x/subspaces`, `x/reports`) 
- There are some ADRs that are related to such modules and are PRs (eg. #616, #640, #624, #639) 
- Based on the definition of AD: 
   > An Architectural Decision (AD) is a software design choice that addresses a functional or non-functional requirement that is **architecturally significant**  

   Some of the merged ADR should have not been ADR in the first place since they are very small changes with no architecture impact ([`ADR-005`](https://github.com/desmos-labs/desmos/blob/master/docs/architecture/adr-005-enabled-snap-shot-by-default.md), [`ADR-009`](https://github.com/desmos-labs/desmos/blob/master/docs/architecture/adr-009-avoid-accidental-dtag-override.md))

For this reason I propose the following things: 

1. Create a new branch called `backlog` where we keep the current `staging` folder
2. Remove the `staging` folder from the `master` branch
3. Re-organize all the ADRs by: 
   1. Closing the currently opened PRs listed above
   2. Deleting the non-ADRs listed above
   3. Re-numbering the merged ADRs 
4. Create one ADR per currently staging module just like Cosmos does before starting to work on that module (subspaces, posts, reports in this order). Each ADR for single module should be created AFTER the dependent module ADR has been merged: 
   ```
   subspaces ADR -> merge -> posts ADR -> merge -> reports ADR
   ```
5. Once the ADR for a module is merged, use the `backlog` branch to get the module implementation, change it to adapt it to the ADR and then merge it inside the "master" branch

What do you guys think?

**Comment by kwunyeung** [comment]
I'm fine with the suggested workflow.

**Comment by dadamu** [comment]
I agree with this workflow.

**Comment by leobragaz** [comment]
I'm okay with this, would you like me to take care of these changes?  @RiccardoM

**Reply by rmontagnin** [reply]
I can take care of this
